epochStart/genesis/genesisBlockCreator.go

```diff
@@ -194,10 +195,14 @@ func createProcessorsForMetaGenesisBlock(
 		Marshalizer:      args.Marshalizer,
 		Uint64Converter:  args.Uint64ByteSliceConverter,
 	}
-	virtualMachineFactory, err := metachain.NewVMContainerFactory(argsHook, args.Economics)
+
+	auctionEnabled := args.Economics.AuctionEnabled()
+	args.Economics.SetAuctionEnabled(false)
```

Comment:
in case of genesis we want to stake directly. Now it is only possible through auction smart contract. So when auction is not enabled, staking is directly called from auction smart contract. If auction is enabled, it is staking does only registration, actual writing to the peerState trie will come after selection.

Reply:
ok